nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -171,16 +174,32 @@ public void writeNow(BufferData buffer) {
 
     void finishRequest() {
         if (keepAlive && connectionQueue != null && socket.isConnected()) {
-            if (connectionQueue.offer(this)) {
+            try {
+                if (connectionQueue.offer(this, QUEUE_TIMEOUT, QUEUE_TIMEOUT_TIME_UNIT)) {
+                    if (LOGGER.isLoggable(DEBUG)) {
+                        LOGGER.log(DEBUG, String.format("[%s] client connection returned %s",
+                                                        channelId,
+                                                        Thread.currentThread().getName()));
+                    }
+                    return;
+                } else {
+                    if (LOGGER.isLoggable(DEBUG)) {
+                        LOGGER.log(DEBUG, String.format("[%s] Unable to return client connection because queue is full %s",
+                                                        channelId,
+                                                        Thread.currentThread().getName()));
+                    }
+                }
+            } catch (InterruptedException ie) {
                 if (LOGGER.isLoggable(DEBUG)) {
-                    LOGGER.log(DEBUG, String.format("[%s] client connection returned %s",
+                    LOGGER.log(DEBUG, String.format("[%s] Unable to return client connection due to '%s' %s",
                                                     channelId,
+                                                    ie.getMessage(),
                                                     Thread.currentThread().getName()));
                 }
-                return;
             }
         }
-        this.release();
+        // Close if unable to add to queue
+        this.close();
```

Comment:
Why the explicit `this` here?

Reply:
was replacing previous this.release, which uses this, but I agree, it is not needed.

Reply:
Removed `this`